### PR TITLE
Created a screen shake option

### PR DESCRIPTION
**Pull request by TheAverageGitHubUser**
Hi, this pull request is about make a screen shake option for pepole who is sensitive to shake

PD:some mods have it, so don't say is useless

### CODE REVIEW

**Comment by DillyzThe1** [comment]
You also need to disable shaking in [FunkinLua.hx](https://github.com/ShadowMario/FNF-PsychEngine/blob/main/source/FunkinLua.hx#L1514).

**Comment by EpokCoolGuy** [comment]
> You also need to disable shaking in [FunkinLua.hx](https://github.com/ShadowMario/FNF-PsychEngine/blob/main/source/FunkinLua.hx#L1514).

thank's i forgot

**Comment by DillyzThe1** [comment]
> > You also need to disable shaking in [FunkinLua.hx](https://github.com/ShadowMario/FNF-PsychEngine/blob/main/source/FunkinLua.hx#L1514).
> 
> thank's i forgot

np

**Comment by DillyzThe1** [comment]
Also, you might want to change your implementation in FunkinLua.hx.
You see, you might cause mods to call a null function, when in reality you want to put _if(ClientPrefs.shaking)_ above _cameraFromString(camera).shake(intensity, duration);_.

Example:

`Lua_helper.add_callback(lua, "cameraShake", function(camera:String, intensity:Float, duration:Float) {
             if(ClientPrefs.shaking)
                        cameraFromString(camera).shake(intensity, duration);
});`

**Comment by Paracosm-Daemon** [comment]
i think you could rename it to "reduced motion" since it's more descriptive about motion + correlates with things like motion sickness more

**Review by MemeHoovy — APPROVED** [review]
I know some mods that use this so this could be useful.

**Review by DaRealDotZ — APPROVED** [review]
Much Needed Option

**Review by Starveldt — APPROVED** [review]

File: source/ClientPrefs.hx [hunk 1]
```diff
@@ -11,6 +11,7 @@ class ClientPrefs {
 	public static var middleScroll:Bool = false;
 	public static var opponentStrums:Bool = true;
 	public static var showFPS:Bool = true;
+        public static var shaking:Bool = true
```
Comment (l1ttleO) [inline source/ClientPrefs.hx:14]:
```suggestion
        public static var shaking:Bool = true;
```

File: source/PlayState.hx [hunk 2]
```diff
@@ -3547,6 +3547,7 @@ class PlayState extends MusicBeatState
 				}
 
 			case 'Screen Shake':
+		if(ClientPrefs.shaking) {
```
Comment (l1ttleO) [inline source/PlayState.hx:3550]:
```suggestion
			if(ClientPrefs.shaking) {
```
some indenting please

File: source/PlayState.hx [hunk 3]
```diff
@@ -3562,6 +3563,7 @@ class PlayState extends MusicBeatState
 						targetsArray[i].shake(intensity, duration);
 					}
 				}
+		}
```
Comment (l1ttleO) [inline source/PlayState.hx:3566]:
```suggestion
			}
```
some indenting please

File: source/FunkinLua.hx [hunk 4]
```diff
@@ -1510,10 +1515,9 @@ class FunkinLua {
 			PlayState.instance.moveCamera(isDad);
 			return isDad;
 		});
-		Lua_helper.add_callback(lua, "cameraShake", function(camera:String, intensity:Float, duration:Float) {
-			cameraFromString(camera).shake(intensity, duration);
-		});
-
+                        Lua_helper.add_callback(lua, "cameraShake", function(camera:String, intensity:Float, duration:Float) {
+			if(ClientPrefs.shaking) cameraFromString(camera).shake(intensity, duration); 
+			});
```
Comment (l1ttleO) [inline source/FunkinLua.hx:1520]:
```suggestion
		Lua_helper.add_callback(lua, "cameraShake", function(camera:String, intensity:Float, duration:Float) {
			if(ClientPrefs.shaking) cameraFromString(camera).shake(intensity, duration); 
		});
```

File: source/options/VisualsUISubState.hx [hunk 5]
```diff
@@ -62,6 +62,13 @@ class VisualsUISubState extends BaseOptionsMenu
 			'bool',
 			true);
 		addOption(option);
+		
+			var option:Option = new Option('Screen Shake',
+			"Uncheck this if you're sensitive to shake!",
+			'shaking',
+			'bool',
+			true);
```
Comment (l1ttleO) [inline source/options/VisualsUISubState.hx:70]:
```suggestion
		var option:Option = new Option('Screen Shake',
		"Uncheck this if you're sensitive to shake!",
		'shaking',
		'bool',
		true);
```

**Review by Octol1ttle — CHANGES_REQUESTED** [review]

**Comment by TheAverageGitHubUser** [comment]
making it for another pull

File: source/PlayState.hx (hunk 2 above)
Comment (TheAverageGitHubUser) [reply source/PlayState.hx:3651]: bruh

**Pull request closed without merging**